generator/docs/NullOperator.md

```diff
@@ -0,0 +1,168 @@
+# `null`, presence, absence and the empty set.
+
+The `null` operator in a profile, expressed as `"is": "null"` or the negated equivalent has several meanings. It can mean (and emit the behaviour) as described below:
+
+| Absence / Presence<br />Described in other technologies as nullability | Field value<br /> |
+| ---- | ---- |
+| _null operator omitted_<br /> **The default**. The field's value may be absent or present | `is null`<br />The field will have _no value_ |
+| `not(is null)`<br />The field's value must be present | `not(is null)`<br />The field must have a value |
+
+Therefore the null operator can:
+- `not(is null)` express fields that must have a value (otherwise known as a non-nullable field)
+- `is null` express fields as having no value (otherwise known as setting the value to `null`)
+- _By omitting the constraint_: express fields as allowing either having a value or the absent of one (otherwise known as a nullable field)
+
+### `null` and interoperability
+`null` is a keyword/term that exists in other technologies and languages, so far as this tool is concerned it relates to the absence or the presence of a value. See [set restriction and generation](SetRestrictionAndGenerarion.md) for more details.
+
+When a field is serialised or otherwise written to a medium, as the output of the generator, it may choose to represent the absence of a value by using the formats' `null` representation, or some other form such as omitting the property and so on.
+
+#### For illustrstion
+CSV files don't support representing the absence of a value differently to an empty string (unless all strings are always wrapped in quotes ([#441](https://github.com/ScottLogic/data-engineering-generator/pull/441)). 
```

Comment:
I suspect the CSV standard doesn't make a distinction between those things, and we shouldn't rely on assigning meaning to distinctions that aren't meaningful in the CSV spec.

Reply:
I agree, and i think this statement, clarifying slightly for completeness.